cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SourceControlManagementHttpHandler.java

```diff
@@ -163,6 +171,38 @@ public void pushApp(FullHttpRequest request, HttpResponder responder,
     }
   }
 
+  /**
+   * Pull the requested application from linked repository and deploy in current namespace.
+   */
+  @POST
+  @Path("/apps/{app-id}/pull")
+  public void pullApp(FullHttpRequest request, HttpResponder responder,
+                      @PathParam("namespace-id") String namespaceId,
+                      @PathParam("app-id") final String appId) throws Exception {
+    checkSourceControlFeatureFlag();
+    ApplicationReference appRef = validateAppReference(namespaceId, appId);
+
+    try {
+      ApplicationRecord appRecord = sourceControlService.pullAndDeploy(appRef);
+      responder.sendJson(HttpResponseStatus.OK, GSON.toJson(appRecord));
+    } catch (NoChangesToPullException | InvalidArtifactException e) {
+      responder.sendString(HttpResponseStatus.BAD_REQUEST, e.getMessage());
+    } catch (ArtifactNotFoundException e) {
+      responder.sendString(HttpResponseStatus.NOT_FOUND, e.getMessage());
+    } catch (ConflictException e) {
+      responder.sendString(HttpResponseStatus.CONFLICT, e.getMessage());
+    } catch (UnauthorizedException e) {
+      responder.sendString(HttpResponseStatus.FORBIDDEN, e.getMessage());
+    } catch (IOException e) {
+      LOG.error("Error reading request body for creating app {}.", appId);
```

Comment:
error logs should include the exception to log the stack trace, otherwise the root cause is lost.

Reply:
oops missed it, seems now we don't need to capture it since there'e no request body. The more explicit exceptions will be caught once https://github.com/cdapio/cdap/pull/14928 is in

btw added back the missed LOG.error `e`  in ApplifecycleHandler